apps/server/lib/lexical/server/project/progress.ex

```diff
@@ -0,0 +1,252 @@
+defmodule Lexical.Server.Project.Progress do
+  require Logger
+
+  defmodule ProgressValue do
+    defstruct [:title, :message, :percentage]
+
+    def new(options) do
+      %__MODULE__{
+        title: options[:title],
+        message: options[:message],
+        percentage: options[:percentage]
+      }
+    end
+  end
+
+  defmodule State do
```

Comment:
all functions in this module should have `%__MODULE__{} = state` as their first parameter.